### PR TITLE
[ESP]fix compilation errors for lock-app(Enable PW RPC) and ipv6only-app with IDF v4.3

### PR DESCRIPTION
**Pull request by wqx6**
#### Problem
Fix compilation errors for lock-app(Enable PW RPC) and ipv6only-app with IDF v4.3 
described in #8076 

#### Change overview
Replace compile options std=c++17 with std=gnu++17 for pw_build.cpp17

#### Testing
Manually verified for the compilation of the two apps.

### CODE REVIEW

**Review by rgoliver — APPROVED** [review]

File: .github/workflows/examples-esp32.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ jobs:
                    .environment/gn_out/.ninja_log
                    .environment/pigweed-venv/*.log
             - name: Build example All Clusters App
-              timeout-minutes: 15
+              timeout-minutes: 20
```
Comment (dhrishi) [inline .github/workflows/examples-esp32.yaml:66]: Why do we need to change/increase the timeout?

**Review by dhrishi — APPROVED** [review]

**Comment by dhrishi** [comment]
@wqx6 Can you please add in the description (Changes overview) about what is done in CMakeLists.txt

**Review by andy31415 — APPROVED** [review]

File: .github/workflows/examples-esp32.yaml (hunk 1 above)
Comment (wqx6) [reply .github/workflows/examples-esp32.yaml]: Because the CI always fails with timeout.

**Review by jelderton — APPROVED** [review]

**Comment by dhrishi** [comment]
@andy31415 @bzbarsky-apple  Can this be merged?

**Review by msandstedt — APPROVED** [review]

**Pull request merged**